plugins/module_utils/network/sonic/config/logging/logging.py

```diff
@@ -287,6 +287,10 @@ def _state_overridden(self, want, have):
         commands = []
         requests = []
 
+        if (have['remote_servers'] == want['remote_servers']) and (have['security_profile'] is None):
+            if 'security_profile' not in want:
+                want['security_profile'] = None
+
```

Comment:
Since 'None' values are not removed in have, the 'security_profile' value can be updated with 'None' as default for the equality check.
```suggestion
        want.setdefault('security_profile', None)
```

Reply:
Addressed